lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -546,7 +549,7 @@ def safely_acquire_lock_for_table(table, &block)
         blocking_transactions = PgHaMigrations::BlockingDatabaseTransactions.find_blocking_transactions("#{PgHaMigrations::LOCK_TIMEOUT_SECONDS} seconds")
         blocking_transactions.any? do |query|
           query.tables_with_locks.any? do |table|
-            target_tables.include?(table)
+            target_tables.include?(table) && table_with_lock.conflicts_with?(table)
```

Comment:
If we make `conflicts_with?` check table equality then we can drop the `include?` here. Right now can't `table_with_lock` be different from `table`?

I think we should rename the arg `table` now to `locked_table`; there are too many "table" variables in scope now, and I had to read this like 3 times.

Does acquiring a lock on a child table mean you necessarily have a (perhaps lower mode) lock on the parent table? I'm trying to figure out here if we need to check against not the requested table but all target tables, and I'm also trying to understand if we might need to check parent tables (I think not, but I'm having a hard time reasoning about it right now).

Reply:
Will have to play around with the `conflicts_with?` change, but definitely agree on the naming. I wrote this code and it's confusing to me so that's probably not a good sign.

> Does acquiring a lock on a child table mean you necessarily have a (perhaps lower mode) lock on the parent table?

I don't think so. Or at least I'm not seeing anything in `pg_locks` that would indicate the parent table gets locked when locking a child partition:

```
pg_ha_migrations_test=# select relation::regclass, locktype, mode from pg_locks;
 relation |  locktype  |      mode
----------+------------+-----------------
 pg_locks | relation   | AccessShareLock
          | virtualxid | ExclusiveLock
(2 rows)

pg_ha_migrations_test=# begin;
BEGIN
pg_ha_migrations_test=*# lock foos3_default;
LOCK TABLE
pg_ha_migrations_test=*# select relation::regclass, locktype, mode from pg_locks;
   relation    |   locktype    |        mode
---------------+---------------+---------------------
 pg_locks      | relation      | AccessShareLock
               | virtualxid    | ExclusiveLock
               | transactionid | ExclusiveLock
 foos3_default | relation      | AccessExclusiveLock
(4 rows)
```